core/remote/change.js

```diff
@@ -20,12 +20,13 @@ export type RemoteFileUpdate = {sideName: 'remote', type: 'FileUpdate', doc: Met
 export type RemoteDirAddition = {sideName: 'remote', type: 'DirAddition', doc: Metadata, was: Metadata}
 export type RemoteDirDeletion = {sideName: 'remote', type: 'DirDeletion', doc: Metadata}
 export type RemoteDirDissociation = {sideName: 'remote', type: 'DirDissociation', doc: Metadata, was: Metadata}
-export type RemoteDirMove = {sideName: 'remote', type: 'DirMove', doc: Metadata, was: Metadata, needRefetch?: true}
+export type RemoteDirMove = {sideName: 'remote', type: 'DirMove', doc: Metadata, was: Metadata, needRefetch?: true, descendantDirMoves?: RemoteDescendantChange[] }
 export type RemoteDirRestoration = {sideName: 'remote', type: 'DirRestoration', doc: Metadata, was: Metadata}
 export type RemoteDirTrashing = {sideName: 'remote', type: 'DirTrashing', doc: Metadata, was: Metadata}
 export type RemoteIgnoredChange = {sideName: 'remote', type: 'IgnoredChange', doc: Metadata|RemoteDoc|RemoteDeletion, detail: string}
 export type RemoteInvalidChange = {sideName: 'remote', type: 'InvalidChange', doc: *, error: Error}
 export type RemoteUpToDate = {sideName: 'remote', type: 'UpToDate', doc: Metadata, was: Metadata}
+export type RemoteDescendantChange = {sideName: 'remote', type: 'DescendantChange', doc: Metadata|RemoteDoc|RemoteDeletion, detail: string}
```

Comment:
`{doc: Metadata, was: Metadata}`?
I assume it doesn't have to be different from the other move subtypes.
Also one can conclude the `details` from the `type` so they're not needed.

Reply:
> Also one can conclude the details from the type so they're not needed.

Actually the type doesn't include the ancestor path, but I wonder whether it's important now that descendant/child moves are pushed into their ancestor. Maybe replace `details` with just `ancestorPath`?

Reply:
Done in #1274